samples/drumthumper/src/main/java/com/plausibleaudio/drumthumper/DrumPlayer.kt

```diff
@@ -62,39 +62,61 @@ class DrumPlayer {
     }
 
     // asset-based samples
-    fun loadWavAssets(assetMgr: AssetManager) {
-        loadWavAsset(assetMgr, "KickDrum.wav", BASSDRUM, PAN_BASSDRUM)
-        loadWavAsset(assetMgr, "SnareDrum.wav", SNAREDRUM, PAN_SNAREDRUM)
-        loadWavAsset(assetMgr, "CrashCymbal.wav", CRASHCYMBAL, PAN_CRASHCYMBAL)
-        loadWavAsset(assetMgr, "RideCymbal.wav", RIDECYMBAL, PAN_RIDECYMBAL)
-        loadWavAsset(assetMgr, "MidTom.wav", MIDTOM, PAN_MIDTOM)
-        loadWavAsset(assetMgr, "LowTom.wav", LOWTOM, PAN_LOWTOM)
-        loadWavAsset(assetMgr, "HiHat_Open.wav", HIHATOPEN, PAN_HIHATOPEN)
-        loadWavAsset(assetMgr, "HiHat_Closed.wav", HIHATCLOSED, PAN_HIHATCLOSED)
+    fun loadWavAssets(assetMgr: AssetManager): Boolean {
+        var allAssetsCorrect = true
+        if (!loadWavAsset(assetMgr, "KickDrum.wav", BASSDRUM, PAN_BASSDRUM)) {
+            allAssetsCorrect = false
```

Comment:
allAssetsCorrect &= loadWavAssets(...) is also an option
More terse but a bit opaque.

Reply:
Wish I had thought of that! (however, Kotlin doesn't have an &&= operator, so a slightly different solution is required).